lib/controllers/project.js

```diff
@@ -21,779 +22,720 @@
+const Http = require('http');
+const Https = require('https');
+const Async = require('async');
+const Uuid = require('node-uuid');
+const Ignore = require('fstream-ignore');
+const Demeteorizer = require('demeteorizer');
+const Packer = require('zip-stream');
+const Through = require('through');
+const Split = require('split');
+
+const Modulus = require('../modulus');
+const Librarian = require('../common/api').librarian;
+const UserConfig = require('../common/api').userConfig;
+const ProgressBar = require('progress');
+const Progress = require('../util/progress');
+const Errors = require('../util/errors');
+const Uploader = require('./uploader');
+
+const TIMEOUT = 1000;
+const PROGRESS_MIN = 0.1;
+const PROGRESS_MAX = 100;
+const PREFIX_SUBSTRING = 8;
+const BYTE_DIV = 1024;
+const NOT_FOUND = 404;
+
+var Project, print = Modulus.io.print;
+Http.globalAgent.maxSockets = Number.MAX_VALUE;
+Https.globalAgent.maxSockets = Number.MAX_VALUE;
+
+// ----------------------------------------------------------------------------
+Project = function () {
 
 };
 
-//-----------------------------------------------------------------------------
-Project.prototype.create = function(name, creatorId, servoSize, imageTagIds, callback) {
+// ----------------------------------------------------------------------------
+Project.prototype.create = function (name, creatorId, servoSize, imageTagIds, callback) {
   var proj = { name: name, creator: creatorId };
   servoSize = parseInt(servoSize, 10);
 
-  if (!isNaN(servoSize)) {
-    proj.servoSize = servoSize;
-  }
-
-  if ('object' === typeof imageTagIds) {
-    proj.imageTagIds = imageTagIds;
-  }
+  if (!isNaN(servoSize)) proj.servoSize = servoSize;
+  if ('object' === typeof imageTagIds) proj.imageTagIds = imageTagIds;
 
-  librarian.project.create(proj, false, userConfig.data.apiKey, callback);
+  Librarian.project.create(proj, false, UserConfig.data.apiKey, callback);
 };
 
-//-----------------------------------------------------------------------------
-Project.prototype.delete = function(projectId, callback) {
-  librarian.project.delete(projectId, userConfig.data.apiKey, function(err, result) {
-    if (err) {
-      return callback(err, null);
-    }
-
-    var dbar = new Progress.indeterminate('Deleting [:bar]');
-    modulus.io.startIndeterminate(dbar);
-
-
-    var checkResult = function() {
-      // Result key should return 'SUCCESS' as the value on successful deletion.
-      if (result.result) {
-        modulus.io.stopIndeterminate(dbar);
-        modulus.io.print(' ');
-        return callback(null, result);
-      } else {
-        setTimeout(checkResult, 1000);
-      }
-    };
+// ----------------------------------------------------------------------------
+Project.prototype.delete = function (projectId, callback) {
+  Librarian.project.delete(
+    projectId, UserConfig.data.apiKey, function (err, result) {
+      var dbar, checkResult;
+      if (err) return callback(err, null);
```

Comment:
Could remove this `null` and the one at 103. Doesn't affect functionality so, IMO, doesn't need to wait until a refactor.